tests/framework_tests/cpp_tests.py

```diff
@@ -0,0 +1,776 @@
+        raised_error = error_handling_tester.test_error_handling['errors_all'][0]
+        assert raised_error['type'] == err_arg['type'], 'Raised error type does not match expected error'
+        assert raised_error['message'] == err_arg['message'], 'Raised error message does not match expected error'
+        assert raised_error['severity'] == err_arg['severity'], 'Raised error severity does not match expected error'
+        assert raised_error == error_handling_tester.test_error_handling['errors_global_all'][0], 'Equal error should be received by global all'
+
+    @pytest.mark.everest_core_config('config-test-cpp-error-handling.yaml')
+    @pytest.mark.asyncio
+    async def test_receive_req_not_sub_error_cleared(
+        self,
+        everest_core: EverestCore,
+        error_handling_tester: ErrorHandlingTester
+    ):
+        """
+        The probe module raises an error of type TestErrorC on the main implementation,
+        then clears this error,
+        TestErrorHandling shouldn't be subscribed to this error type.
+        Checks that the error_cleared is subscribed correctly.
+        """
+
+        everest_core.start(standalone_module='probe')
+        err_arg, uuid = error_handling_tester.probe_module_main_raise_error_c()
+        await asyncio.sleep(0.5)
+        error_handling_tester.probe_module_main_clear_error_by_uuid(uuid)
+        await asyncio.sleep(0.5)
+        expected_state = ErrorHandlingTesterState(
+            len_test_error_handling_errors_all=1,
+            len_test_error_handling_errors_global_all=1,
+            len_test_error_handling_errors_cleared_all=1,
+            len_test_error_handling_errors_cleared_global_all=1
+        )
+        assert error_handling_tester.get_state() == expected_state, 'State does not match expected state'
+        raised_error = error_handling_tester.test_error_handling['errors_all'][0]
+        cleared_error = error_handling_tester.test_error_handling['errors_cleared_all'][0]
+        assert raised_error == cleared_error, 'Raised error does not match cleared error'
+        assert cleared_error == error_handling_tester.test_error_handling['errors_global_all'][0], 'Equal error should be received by global all'
+
+
+    @pytest.mark.everest_core_config('config-test-cpp-error-handling.yaml')
+    @pytest.mark.asyncio
+    async def test_receive_not_req_error(
+        self,
+        everest_core: EverestCore,
+        error_handling_tester: ErrorHandlingTester
+    ):
+        """
+        The probe modules triggers the test_error_handling_not_req module
+        to raise an error of type TestErrorA. This implementation is not required
+        by TestErrorHandling, so it shouldn't be subscribed to this error type.
+        Checks that the error is subscribed correctly.
+        """
+
+        everest_core.start(standalone_module='probe')
+        err_arg, uuid = await error_handling_tester.test_error_handling_not_req_raise_error_a()
```

Comment:
This test case is why a third module is needed.
The current design of error managment and probe modules doesn't allow/enable the probe_module to raise errors on a unrequired implementation